cmd/mps-control-daemon/main.go

```diff
@@ -168,9 +172,16 @@ func startDaemons(c *cli.Context, cfg *Config) ([]*mps.Daemon, bool, error) {
 	}
 	spec.DisableResourceNamingInConfig(logger.ToKlog, config)
 
+	nvmllib := nvml.New()
+	devicelib := device.New(device.WithNvml(nvmllib))
+	infolib := nvinfo.New(
+		nvinfo.WithNvmlLib(nvmllib),
+		nvinfo.WithDeviceLib(devicelib),
+	)
+
 	// Update the configuration file with default resources.
 	klog.Info("Updating config with default resource matching patterns.")
-	err = rm.AddDefaultResourcesToConfig(config)
+	err = rm.AddDefaultResourcesToConfig(infolib, nvmllib, devicelib, config)
```

Comment:
Does it make sense to introduce a new struct to hold all of these instead of as individual parameters?

Reply:
I think I would like to do this as a follow-up if you're ok with that.